pkg/pool-utils/test/ExternalCallLib.test.ts

```diff
@@ -0,0 +1,63 @@
+import { Contract } from 'ethers';
+
+import { deploy } from '@balancer-labs/v2-helpers/src/contract';
+import { expect } from 'chai';
+import { solidityKeccak256 } from 'ethers/lib/utils';
+
+describe('ExternalCallLib', function () {
+  let maliciousReverter: Contract;
+  let caller: Contract;
+
+  sharedBeforeEach(async function () {
+    maliciousReverter = await deploy('MaliciousQueryReverter');
+    caller = await deploy('MockExternalCaller', { args: [maliciousReverter.address] });
+  });
+
+  describe('when calling into a malicious contract in a swap', () => {
+    context('when call is protected', () => {
+      it('reverts with MALICIOUS_QUERY_REVERT', async () => {
+        await expect(caller.protectedSwapExternalCall()).to.be.revertedWith('MALICIOUS_QUERY_REVERT');
+      });
+    });
```

Comment:
We're missing tests where we check we bubble up the original revert reason for non malicious callees. The bubble up tests we have here are all of unprotected calls, meaning our bubble up code is not being executed.